Detectors/TRD/base/test/testDigit.cxx

```diff
@@ -0,0 +1,185 @@
+  Digit g(0, 0, NCOLMCM - 1); // row 0 pad 17 --- should be mcm 0 and channel 2
+  testDigitDetROBMCM(g, 0, 0, 0, 2);
+
+  Digit h(0, 0, 0, 2);
+  testDigitDetPadRow(h, 0, 0, NCOLMCM - 1);
+
+  //test block2 repeat block1 but at the edge of a rob boundary i.e. going from row0 the 72nd pad to 73rd. Spanning the half of 144(NCOLUMN)
+  Digit i(0, 0, (NCOLUMN / 2) - 1);
+  testDigitDetROBMCM(i, 0, 0, 3, 2);
+  //check the reverse creation
+  Digit k(0, 0, 3, 2);
+  testDigitDetPadRow(k, 0, 0, (NCOLUMN / 2) - 1);
+
+  Digit j(0, 0, NCOLUMN / 2);
+  testDigitDetROBMCM(j, 0, 1, 0, 19);
+  //check the reverse creation
+  Digit l(0, 1, 0, 19);
+  testDigitDetPadRow(l, 0, 0, NCOLUMN / 2);
+
+  // now repeat the same for another part of the first chamber, middle rows
+  //
+  Digit m(0, 12, (NCOLUMN / 2) - 1);
+  testDigitDetROBMCM(m, 0, 6, 3, 2);
+  //check the reverse creation
+  Digit n(0, 6, 3, 2);
+  testDigitDetPadRow(n, 0, 12, (NCOLUMN / 2) - 1);
+
+  Digit o(0, 12, NCOLMCM - 1);
+  testDigitDetROBMCM(o, 0, 6, 0, 2);
+  Digit p(0, 6, 0, 2);
+  testDigitDetPadRow(p, 0, 12, NCOLMCM - 1);
+
+  //and now for the last row.
+  Digit q(0, 15, (NCOLUMN / 2) - 1);
+  testDigitDetROBMCM(q, 0, 6, 15, 2);
+  //check the reverse creation
+  Digit r(0, 6, 3, 2);
+  testDigitDetPadRow(r, 0, 12, (NCOLUMN / 2) - 1);
+
+  Digit s(0, 15, NCOLMCM - 1);
+  testDigitDetROBMCM(s, 0, 6, 12, 2);
+  Digit t(0, 6, 0, 2);
+  testDigitDetPadRow(t, 0, 12, NCOLMCM - 1);
+
+  // as a last check that for detector changes.
+  //
+  Digit u(1, 15, (NCOLUMN / 2) - 1);
+  testDigitDetROBMCM(u, 1, 6, 15, 2);
+  //check the reverse creation
+  Digit v(1, 6, 3, 2);
+  testDigitDetPadRow(v, 1, 12, (NCOLUMN / 2) - 1);
+
+  Digit w(10, 15, NCOLMCM - 1);
+  testDigitDetROBMCM(w, 10, 6, 12, 2);
+  Digit x(10, 6, 0, 2);
+  testDigitDetPadRow(x, 10, 12, NCOLMCM - 1);
+
+  /* 
+  * The below is left in, it helps to remove confusion when debugging
+ for(int rob=0;rob<8;rob++)for(int mcm=0;mcm<16;mcm++)for(int channel=0;channel<21;channel++){
+  std::cout << "Digit e(0,"<<rob<<"," << mcm <<","<< channel<<");" << std::endl;
+  Digit e(0,rob,mcm,channel);
+    std::cout << " e is " << e.getRow() << " " << e.getPad();
+    std::cout << " for an rob:mcm combo of " << e.getROB() << ":"<< e.getMCM() << " and adc channel:" << e.getChannel() <<std::endl;
+  std::cout << "Digit f(0,e.getRow(),e.getPad())" << std::endl;;
+  Digit f(0,e.getRow(),e.getPad());
+    std::cout << " f is " << f.getRow() << " " << f.getPad();
+    std::cout << " for an rob:mcm combo of " << f.getROB() << ":"<< f.getMCM() << " and adc channel:" << f.getChannel() <<std::endl;
+    std::cout << "*********************************************************************" << std::endl;
+  }
+  */
+
+  //now check that the timebins get correctly assigned on instantiation
+  ArrayADC data;
+  std::iota(data.begin(), data.end(), 42); // 42 for my personal amusement.
+  Digit z(10, 15, NCOLMCM - 1, data);
+  testDigitDetROBMCM(w, 10, 6, 12, 2);
```

Comment:
It should be `z` instead of `w`.

Reply:
fixed, z and w are the same det/rob/mcm combination.